props/saber_fett263_buttons.h

```diff
@@ -285,47 +440,1090 @@ EFFECT(faston); // for EFFECT_FAST_ON
+      f->skipwhite();
+      f->readVariable(action);
+      if (!strcmp(action, "clash")) {
+          if (SFX_clsh) {
+             file = SFX_clsh.get_min_file();
+          } else {
+             file = SFX_clash.get_min_file();
+          }
+        clash_rec[i].stance = STANCE_CLASH;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        continue;
+      }
+      if (!strcmp(action, "lockup")) {
+          if (SFX_lock) {
+            file = SFX_lock.get_min_file();
+          } else {
+            file = SFX_lockup.get_min_file();
+          }
+        clash_rec[i].stance = STANCE_LOCKUP;
+        f->skipspace();
+        clash_rec[i].sound_number1 = (int)(f->readFloatValue() - SFX_bgnlock.get_min_file());
+        f->skipspace();
+        clash_rec[i].sound_number2 = (int)(f->readFloatValue() - file);
+        f->skipspace();
+        clash_rec[i].sound_number3 = (int)(f->readFloatValue() - SFX_endlock.get_min_file());
+        f->skipspace();
+        continue;
+      }
+      clash_rec[i].stance = STANCE_END;
+      return ReadStatus:: READ_END;
+      break;
+    }
+  }
+
+  void SetVariable(const char* variable, float v) override {
+  }
+
+  enum FormStance {
+    STANCE_END = 0,
+    STANCE_CLASH,
+    STANCE_LOCKUP,
+  };
+
+  struct FormEntry {
+    FormStance stance;
+    int sound_number1;
+    int sound_number2;
+    int sound_number3;
+  };
+
+  FormEntry clash_rec[32];
+ 
+};
+#endif
+
+#ifdef FETT263_EDIT_MODE_MENU
+// Edit Length 
+// NewLength gets length value from menu for use in LengthEdit
+class NewLength {
+public:
+  static int length_;
+  static void EditLength(int length) {
+    length_ = length - 1;
+    return;
+  }
+  static int GetNewLength() {
+    return length_;
+  }
+};
+
+int NewLength::length_ = 0;
+
+// Edit Length Preview for Edit Mode
+// LengthEdit uses blade color and creates single white pixel at last pixel from NewLength, black above last pixel
+template<class BASE, class LIGHTUP, class BLACK = BLACK>
+class LengthEdit : protected NewLength {
+public:
+  void run(BladeBase* blade) {
+    base_.run(blade);
+    lightup_.run(blade);
+    black_.run(blade);
+    led_ = NewLength::GetNewLength();
+  }
+
+  OverDriveColor getColor(int led) {
+    if (led == led_) return lightup_.getColor(led);
+    if (led > led_) return black_.getColor(led);
+    return base_.getColor(led);
+  }
+private:
+  int led_;
+  BASE base_;
+  LIGHTUP lightup_;
+  BLACK black_;
+};
+
+// Edit Style Settings
+class NewInt {
+  public: 
+    static int new_value_;
+    static void EditInt(int new_value) {
+      new_value_ = new_value;
+    }
+    static int GetNewInt() {
+      return new_value_;
+    }
+};
+
+int NewInt::new_value_;
+
+class IntEdit : protected NewInt {
+  public:
+    int int_edit_;
+    void run(BladeBase* blade) {
+    int_edit_ = NewInt::GetNewInt();
+    }
+    int getInteger(int led) { return int_edit_; }    
```

Comment:
If we have a global variable instead of NewInt, then we can just return that here.
Or. We could merge NewInt and IntEdit together and make int_edit_ the static variable.

Reply:
needed the void run so I use the static int and return int_edit, would not compile without.